hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineDiffHelper.java

```diff
@@ -144,44 +172,54 @@ private static List<Pair<HoodieInstant, HoodieInstant>> getPendingCompactionTran
    */
   public static class TimelineDiffResult {
 
-    private final List<HoodieInstant> newlySeenInstants;
+    private final List<HoodieInstant> newlySeenCompletedAndRewriteInstants;
     private final List<HoodieInstant> finishedCompactionInstants;
```

Comment:
so we are not not interested in removed Compaction instants is it? curious to know why this is diff compared to log compactions.

Reply:
Ideally compaction are not removed, since after creating a compaction plan even though they are in inflight state new data is written to new file slices. So, if we were to remove compaction, then we might need to remove latest files. We could add APIs to remove them, but it is bit risky.
Also compaction plans are immutable plans so writers cannot delete them, only developers can remove them manually by using restore operations.